### PR TITLE
Support users can edit application phone numbers

### PR DESCRIPTION
**Pull request by duncanjbrown**
Add controller, form, route and view for editing applicant details. Only
editing phone number is supported (so far).

## Context

Easier for support staff and devs.

## Changes proposed in this pull request

Screenshot 2020-11-17 at 17 54 15

Screenshot 2020-11-17 at 17 54 06

Screenshot 2020-11-17 at 18 03 48

## Guidance to review

a) Does it work properly?
b) I think it's ok for one field, but I don't really like adding the "Change" link to the shared component. Would rather decorate somehow? Ideas welcome.

## Link to Trello card

https://ukgovernmentdfe.slack.com/archives/CQA64BETU/p1605606666428400

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/personal_details_component.rb [hunk 1]
```diff
@@ -51,10 +52,19 @@ def email_row
   end
 
   def phone_number_row
-    {
+    basic_row = {
       key: 'Phone number',
       value: phone_number || MISSING,
     }
+
+    if current_namespace == 'support_interface'
```
Comment (tijmenb) [inline app/components/personal_details_component.rb:60]: I think we should duplicate this component if we're making them namespace-specific. It's already a bit questionable that we share logic-heavy components between namespaces (see https://github.com/DFE-Digital/apply-for-teacher-training/pull/2071 for some considerations).

We could make the duplication less severe by pushing some stuff to the model.

File: app/controllers/support_interface/application_forms/applicant_details_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module SupportInterface
+  module ApplicationForms
+    class ApplicantDetailsController < SupportInterfaceController
```
Comment (tijmenb) [inline app/controllers/support_interface/application_forms/applicant_details_controller.rb:3]: Thoughts behind "Applicant"? Perhaps otherwise `CandidateDetailsController` or `EditApplicationFormController` would be more in line with the other code.

File: app/controllers/support_interface/application_forms/applicant_details_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          ApplicationForm.find(params[:application_form_id]),
+        )
+      end
+
+      def update
+        @details = EditApplicantDetailsForm.new(
+          ApplicationForm.find(params[:application_form_id]),
+        )
+
+        @details.phone_number = edit_application_params[:phone_number]
```
Comment (tijmenb) [inline app/controllers/support_interface/application_forms/applicant_details_controller.rb:15]: This might work:

```suggestion
        @details.assign_attributes(edit_application_params)
```

It removes the duplication with `edit_application_params`, so that if we add new attributes they only need to be added once.

File: app/views/support_interface/application_forms/applicant_details/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @details, url: support_interface_application_form_update_applicant_details_path do |f| %>
+      <%= f.govuk_error_summary %>
+
+      <%= f.govuk_fieldset legend: { text: 'Edit applicant details', size: 'l' } do %>
+        <%= f.govuk_text_field :phone_number, label: { text: 'Phone number' } %>
+      <% end %>
+
```
Comment (tijmenb) [inline app/views/support_interface/application_forms/applicant_details/edit.html.erb:12]: So I don't forget: we could add the audit comment field here where the support user can paste in a link to Zendesk, so that we have context on why the change was made.

File: config/locales/en.yml [hunk 5]
```diff
@@ -212,6 +212,11 @@ en:
           attributes:
             further_conditions:
               too_long: "%{name} must be %{limit} characters or fewer"
+        support_interface/application_forms/edit_applicant_details_form:
```
Comment (tijmenb) [inline config/locales/en.yml:215]: Wondering if there's a way to dedupe this. Perhaps adding the text in `PhoneNumberValidator`? Not a blocker.

**Review by tijmenb — COMMENTED** [review]
Very nice! This will be so useful.

File: app/components/personal_details_component.rb (hunk 1 above)
Comment (duncanjbrown) [reply app/components/personal_details_component.rb:60]: Agreed!

File: app/controllers/support_interface/application_forms/applicant_details_controller.rb (hunk 2 above)
Comment (duncanjbrown) [reply app/controllers/support_interface/application_forms/applicant_details_controller.rb:3]: I followed the UI here. No particularly strong feelings though.

File: app/views/support_interface/application_forms/applicant_details/edit.html.erb (hunk 4 above)
Comment (duncanjbrown) [reply app/views/support_interface/application_forms/applicant_details/edit.html.erb:12]: Very good idea, will add!

**Comment by duncanjbrown** [comment]
@tijmenb Now with added duplication!

File: app/views/support_interface/application_forms/applicant_details/edit.html.erb (hunk 4 above)
Comment (duncanjbrown) [reply app/views/support_interface/application_forms/applicant_details/edit.html.erb:13]: Screenshot 2020-11-18 at 10 45 00

**Review by tijmenb — APPROVED** [review]
🚀 🚀 🚀
